src/Eori/EoriValidator.php

```diff
@@ -2,21 +2,85 @@
 
 namespace CrazyFactory\Validation\Eori;
 
-use CrazyFactory\HarmonizedSystem\Countries;
+use SoapClient;
 
 class EoriValidator
 {
     /**
+     * @var SoapClient
+     */
+    private $soapClient;
+
+    /**
+     * Interpret the result as valid if match these fault codes
+     * @var array
+     */
+    private $ignoreFaultCodes = [];
+
+    /**
+     * Interpret the result as invalid if match these fault codes
+     * @var array
+     */
+    private $negativeFaultCodes = [];
+
+    public function __construct(SoapClient $client = null)
+    {
+        $this->soapClient = $client ?? new SoapClient('https://ec.europa.eu/taxation_customs/dds2/eos/validation/services/validation?wsdl');
```

Comment:
shouldnt be in construct.

should be instantiated only when we call `validate()` and is about to hit the endpoint